ecl_type_traits/include/ecl/type_traits/numeric_limits.hpp

```diff
@@ -53,7 +53,7 @@ namespace ecl {
  * @tparam T : the type to check (currently must be a fundamental type).
  */
 template <typename T>
-class ecl_type_traits_PUBLIC numeric_limits : public std::numeric_limits<T> {
+class numeric_limits : public std::numeric_limits<T> {
```

Comment:
I think you need to add these back in, as people who use these classes won't link.